src/main/java/dev/jbang/Util.java

```diff
@@ -531,36 +531,63 @@ public static String getStableID(String input) {
 	private static String extractFileFromGist(String url) {
 
 		try {
-			String gistapi = url.replaceFirst("^https://gist.github.com/(([a-zA-Z0-9]*)/)?(?<gistid>[a-zA-Z0-9]*)$",
+			String[] pathPlusAnchor = url.split("#");
+			String fileName = "";
+			if (pathPlusAnchor.length == 2) {
+				String[] anchor = pathPlusAnchor[1].split("-");
+				if (anchor.length < 2)
+					throw new IllegalArgumentException("Invalid Gist url: " + url);
+				fileName = anchor[1];
+				for (int i = 2; i < anchor.length - 1; ++i)
+					fileName += "-" + anchor[i];
+			}
+			String gistapi = pathPlusAnchor[0].replaceFirst(
+					"^https://gist.github.com/(([a-zA-Z0-9]*)/)?(?<gistid>[a-zA-Z0-9]*)$",
 					"https://api.github.com/gists/${gistid}");
-			// Util.info("looking at " + gistapi);
+
+			Util.verboseMsg("looking at " + gistapi);
 			String strdata = null;
 			try {
 				strdata = readStringFromURL(gistapi);
 			} catch (IOException e) {
-				// Util.info("error " + e);
+				Util.verboseMsg("error " + e);
```

Comment:
if you want to include it provide some context for the user, like Util.verboseMsg("Error when extracting file from gist url: ",e);